src/utils/dataMatchesContraints.js

```diff
@@ -19,6 +19,12 @@ module.exports = function(data, constraints) {
     const valueType = typeof value;
     if (mandatory) {
       return !value || valueType !== type || (supportedValues && !supportedValues.includes(value));
+    } else {
+      if (value) {
+        return valueType !== type || (supportedValues && !supportedValues.includes(value));
```

Comment:
you could replace lines from 23 to 27 by:
`return value && (valueType !== type || (supportedValues && !supportedValues.includes(value)));`

Reply:
fixed in https://github.com/adobe/adobe-client-data-layer/pull/79/commits/6f67c189ce33d966b41807634d113d9c53a75db2